src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,90 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct financial_tooltip_info
+{
+    const CoordsXY coords;
+    const money32 money;
+};
+
+static const auto chartMaxWidth = 63 * 6;
+static const auto chartMaxHeight = 164;
+
+static int32_t indexForCursorAndHistory(const int32_t& historyCount, const int32_t& cursorX, const int32_t& chartX)
+{
+    const auto cursorOffset = 3;
+    const auto offsettedCursorX = cursorX + cursorOffset;
+    return (historyCount - 1) - (offsettedCursorX - chartX) / 6;
+}
+
+static const CoordsXY coordsForHistoryIndex(
+    const int32_t& index, const money32* history, const int32_t& chartX, const int32_t& chartY, const int32_t& modifier,
+    const int32_t& offset)
+{
+    CoordsXY coords;
+    coords.x = chartX + 6 * (63 - index);
+    coords.y = chartY + 170 - 6 - ((((history[index] >> modifier) + offset) * 170) / 256);
+    return coords;
+}
+
+static const financial_tooltip_info finance_tooltip_info_from_money(
+    const money32* history, const int32_t& historyCount, const int32_t& modifier, const int32_t& offset,
+    const MapRange& chartFrame, const ScreenCoordsXY& cursorPosition)
```

Comment:
Seems odd to use `MapRange` for something that isn't the map.

Reply:
Are there better structs for this kind of abstraction?

Reply:
Hmm I don't think so, but you are welcome to make a `Rect` struct.